Create self-hosted-subscriptions (#5067)
mattermost/docs: source/about/licensing-and-subscription.rst

> Licensing and Subscription

Reviewer: Given that the title of this existing docs page has changed, let's also change the filename of the file itself to match, and add a redirect from the existing page to the new page.

Author: Updated to just 'subscription.rst'

Reviewer: Thanks! Please also update the LHS link to the updated file via ``overview/index.html`` like this:

```
.. toctree::
   :maxdepth: 2
   :glob:

   /about/product
   ...
   /about/subscription
```

Then, add the following redirect link in ``conf.py``, under the ``# Redirects section``

```
"about/licensing-and-subscription": "https://docs/mattermost.com/about/subscription.html",
```

Feel free to make this the very first redirect if you like since it starts with "a".

Author: Added and updated

Subscription Overview
=====================

> Mattermost offers features through `Starter, Professional, or Enterprise subscription plans <www.mattermost.com/pricing>`__. Your subscription to a plan determines what features you have access to. When choosing a subscription, you can choose the deployment type - self-hosted or cloud, and the plan - Starter, Professional, or Enterprise. 

Reviewer: ```suggestion
Mattermost offers features through `Starter, Professional, or Enterprise subscription plans <www.mattermost.com/pricing>`__. Your plan subscription determines what features you have access to. When choosing a subscription, you can choose the deployment type as self-hosted or cloud, and the plan as Starter, Professional, or Enterprise. 
```

> Mattermost self-hosted requires a license to be applied to access features in the Professional or Enterprise plans. Mattermost offers a `secure self-service Customer Portal <https://customers.mattermost.com>`__ where you can easily purchase and manage your Mattermost self-hosted subscriptions. When you purchase a subscription, a license is generated. Please see :ref:`self-hosted-subscriptions` for more details. 

Reviewer: ```suggestion
Mattermost self-hosted deployments require a license to be applied to access features in the Professional or Enterprise plans. Mattermost offers a `secure self-service Customer Portal <https://customers.mattermost.com>`__ where you can easily purchase and manage your Mattermost self-hosted subscriptions. When you purchase a subscription, a license is generated. Please see :ref:`self-hosted-subscriptions` for more details. 
```

> Mattermost Cloud is a software-as-as-service, and you can simply sign up and start using Mattermost in a trial mode and upgrade to the edition you desire within the product. Please see :ref:`cloud-subscriptions` for more details.

Reviewer: ```suggestion
Mattermost Cloud is a software-as-as-service deployment. You can simply sign up and start using Mattermost in a trial mode, then upgrade to the edition you desire within the product. Please see :ref:`cloud-subscriptions` for more details.
```

Author: Applied in new file created when I changed the file name.

> For more general information and frequently asked questions on licensing and subscriptions, please see below. 

Reviewer: ```suggestion
For more general information and frequently asked questions on licensing and subscriptions, please see the following sections. 
```

Author: Applied in new file created when I changed the file name.

In this case, the "physical address" is the location where the individual, who will be receiving the license key, is physically located and will be able to access the software for installation.

How is a user defined for subscriptions?

> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

    
    Mentions for AD/LDAP groups are not shown in the autocomplete menu.
    

>     Group mentions are no longer highlighted in text and do not trigger new notifications.", "Use the `CLI <https://docs.mattermost.com/manage/command-line-tools.html#mattermost-group>`_ to modify group sync settings for the team/channel."

Reviewer: We'll want to include the mmctl equivalent here as well

Author: Applied in new file created when I changed the file name.

    "High Availability", "High Availability is disabled. If all nodes in a cluster continue running, the nodes will stop communicating and caches will get out of sync. This is likely to cause delays in messages, notifications, etc.", "None needed."
    "Performance monitoring", "Monitoring is disabled and Grafana will no longer update with new data.", "None needed."
    "Compliance exports", "Jobs are no longer scheduled in the job server. Data is not exported.", "None needed."
    "Data retention", "Jobs are no longer scheduled in the job server. Data is not deleted.", "None needed."
    "Custom terms", "Custom terms no longer displayed to end users on login. Data is retained in the Terms of Service database table.", "None needed."
    "Custom announcement banners", "No longer visible and is replaced by the default announcement banner.", "None needed."
    "Multi-factor authentication (MFA)", "MFA is no longer enforced/required for new accounts but remains enabled for those who configured it.", "None needed."

>     "Permissions", "Permissions are retained in the database in a frozen state and cannot be modified in the System Console.", "Use the `CLI https://docs.mattermost.com/manage/command-line-tools.html#mattermost-permissions>`_ to reset permissions to default."

Reviewer: ```suggestion
    "Permissions", "Permissions are retained in the database in a frozen state and cannot be modified in the System Console.", "Use the `CLI <https://docs.mattermost.com/manage/command-line-tools.html#mattermost-permissions>`__ to reset permissions to default."
```

We'll want to include the mmctl equivalent here as well

Author: `mmctl <https://docs.mattermost.com/manage/mmctl-command-line-tool.html#mmctl-permissions>`__ added in new file.

    "Guest accounts", "Guests that are not actively logged in are prevented from logging in. Guests who are actively logged in are able to use Mattermost until their session expires or they log out.", "None needed."
    
Do you have a program for official non-profits, open source projects, and charities?
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Yes. The Mattermost Nonprofit License enables foundations and other nonprofit organizations that support open source projects or other technology initiatives to apply the benefits of Mattermost Enterprise Edition to advancing their missions with special nonprofit pricing.

> To be eligible, an organization needs to be an official nonprofit or charity; as well as non-government, non-academic and non-commercial in nature, with no religious affiliation; and that would otherwise be unable to afford the commercial version of Mattermost software. If your organization doesn’t fit this description, we suggest that you purchase a `commercial license <https://mattermost.com/pricing-self-managed/>`__ instead.

Reviewer: the self-managed URL likely needs to change

Author: updated to general mattermost.com/pricing

Organizations that receive a Mattermost Nonprofit License must make their server publicly-accessible for anyone to sign up and join. Mattermost will also have the right to place the name and logo of the nonprofit or charitable institution on our website.

> To apply for the Mattermost Nonprofit License, `please complete this form <https://docs.google.com/spreadsheets/d/1aEpjFLHcrSpJF3wfi2NItevXtYpZUOSbge9o_pLagXU/edit#gid=446002361>`__, or contact us at community[at]mattermost.com.

Reviewer: Only Mattermost staff can currently access this form.

also: consider updating the contact us address to be a mailto link

Author: This link is different in the live site currently.  Updated to reflect the correct form. Need some help with the mailto formatting

Do you have discounted subscriptions for academic institutions?
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Yes, for academic institutions we offer `Mattermost Enterprise Edition Standard <https://mattermost.com/education/>`_ for no charge to students (staff members pay the regular price). You need to pay for at least 10 staff members in order to qualify for an academic license. Please see `Mattermost Academic Licensing <https://docs.google.com/forms/d/e/1FAIpQLSfdl9fTwahgMQu0hb65A58OWzzR3541VwU-MbT0f3y1ND4QhA/viewform>`_ for more information.

Reviewer: The google form from this link includes E10 references

Author: Asked Jason who owns it to get it queued for updates.
